src/swap.ts

```diff
@@ -48,34 +47,35 @@ export class Swap extends Core {
     outputBlindingKeys,
   }: requestOpts): Promise<Uint8Array> {
     // Check amounts
-    const msg = new proto.SwapRequest();
-    msg.setId(makeid(8));
-    msg.setAmountP(amountToBeSent);
-    msg.setAssetP(assetToBeSent);
-    msg.setAmountR(amountToReceive);
-    msg.setAssetR(assetToReceive);
-    msg.setTransaction(psetBase64);
+    const msg = proto.SwapRequest.create({
+      id: makeid(8),
+      amountP: BigInt(amountToBeSent),
+      assetP: assetToBeSent,
+      amountR: BigInt(amountToReceive),
```

Comment:
the Swap/Trade API parameters should'nt move from `number` to `bigInt` (like amountToSend, amountToReceive members etc...) ? Not sure to get the need behing bigint, can u elaborate this point, my intuition is because `Number.MAX_SAFE_INTEGER` does not cover the max satoshis amount ?

Reply:
Yes, now that native js BigInt has a good browser support I think it's a good idea to use it (https://blog.logrocket.com/why-and-how-to-use-bigint/). It's the default option on protobuf-ts but we can also use Number. The thing is when we compile stubs with BigInt it's all or nothing.

Reply:
What about splitting this change in a dedicated PR? so we also have time to see if does not break apps/clients?

I would keep `number` for now

Reply:
Ok lets do that

Reply:
or it seems people are ovecoming this adding a flag in protos about js types

https://github.com/koinos/koinos-proto/pull/23/files

Fun we never had problem till now 🤔